authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -118,27 +128,319 @@ public List<String> policyResourceDefinesRule() {
+    policy.getRowFilterPolicyItems().forEach(RangerHelper::checkPolicyItemAccess);
+    policy.getDataMaskPolicyItems().forEach(RangerHelper::checkPolicyItemAccess);
+
+    return policy;
+  }
+
+  @Override
+  /** Wildcard search the Ranger policies in the different Ranger service. */
+  protected List<RangerPolicy> wildcardSearchPolies(
+      AuthorizationMetadataObject authzMetadataObject) {
+    Preconditions.checkArgument(authzMetadataObject instanceof PathBasedMetadataObject);
+    PathBasedMetadataObject pathBasedMetadataObject = (PathBasedMetadataObject) authzMetadataObject;
+    List<String> resourceDefines = policyResourceDefinesRule();
+    Map<String, String> searchFilters = new HashMap<>();
+    searchFilters.put(SearchFilter.SERVICE_NAME, rangerServiceName);
+    resourceDefines.stream()
+        .forEach(
+            resourceDefine -> {
+              searchFilters.put(
+                  SearchFilter.RESOURCE_PREFIX + resourceDefine,
+                  getAuthorizationPath(pathBasedMetadataObject));
+            });
+    try {
+      return rangerClient.findPolicies(searchFilters);
+    } catch (RangerServiceException e) {
+      throw new AuthorizationPluginException(e, "Failed to find the policies in the Ranger");
+    }
+  }
+
+  /**
+   * IF rename the SCHEMA, Need to rename these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` <br>
+   * IF rename the TABLE, Need to rename these the relevant policies, `{schema}.*`, `{schema}.*.*`
+   * <br>
+   */
+  @Override
+  protected void doRenameMetadataObject(
+      AuthorizationMetadataObject authzMetadataObject,
+      AuthorizationMetadataObject newAuthzMetadataObject) {
+    Preconditions.checkArgument(
+        authzMetadataObject instanceof PathBasedMetadataObject,
+        "The metadata object must be a PathBasedMetadataObject");
+    Preconditions.checkArgument(
+        newAuthzMetadataObject instanceof PathBasedMetadataObject,
+        "The metadata object must be a PathBasedMetadataObject");
+    updatePolicyByMetadataObject(
+        newAuthzMetadataObject.type().metadataObjectType(),
+        authzMetadataObject,
+        newAuthzMetadataObject);
+  }
+
+  @Override
+  protected void updatePolicyByMetadataObject(
+      MetadataObject.Type operationType,
+      AuthorizationMetadataObject oldAuthzMetaobject,
+      AuthorizationMetadataObject newAuthzMetaobject) {
+    PathBasedMetadataObject newPathBasedMetadataObject =
+        (PathBasedMetadataObject) newAuthzMetaobject;
+    List<RangerPolicy> oldPolicies = wildcardSearchPolies(oldAuthzMetaobject);
+    List<RangerPolicy> existNewPolicies = wildcardSearchPolies(newAuthzMetaobject);
+    if (oldPolicies.isEmpty()) {
+      LOG.warn("Cannot find the Ranger policy for the metadata object({})!", oldAuthzMetaobject);
+      return;
+    }
+    if (!existNewPolicies.isEmpty()) {
+      LOG.warn("The Ranger policy for the metadata object({}) already exists!", newAuthzMetaobject);
```

Comment:
I wonder if it's a normal case that `existNewPolicies` is NOT empty?

Reply:
Yes, that could be the case.
Maybe a user add a policy in the Ranger manually.